### DISCUSSION TITLE
Next.js API route works locally, but not on Vercel?

### DISCUSSION THREAD

**Discussion by christina-de-martinez** [vercel/community#2020] [Help] [answered]
### Summary

I have a Next.js web app that I'm hosting on Vercel. I wanted to use the ImageKit API to load and display images on the site. In order to do so and bypass CORS issues, I needed to have the request on the same origin as the web app. So I'm making a Next.js API call as a sort of proxy, if you will. 

This solved my CORS issues locally. The API route works perfectly locally, both on production and dev builds. But when I publish it to Vercel, it stops working, and I get CORS errors. 

I tried completely disabling CORS on my Next.js API, like the steps suggest [here](https://vercel.com/guides/how-to-enable-cors), and it didn't help.

Has anyone run into anything similar? Am I just totally thinking about this wrong? 

### Additional information

This is the entire contents of /pages/api/imageKit.js:

```
import axios from "axios";

export default async (req, res) => {
    const query = req.query;
    const { shopId } = query;

    const imageKitBaseUrl = "https://api.imagekit.io/v1";
    const basicAuth = `Basic ${process.env.NEXT_PUBLIC_IMAGEKIT_PRIVATE_KEY}`;
    const imageKitUrl = `${imageKitBaseUrl}/files?path=/shops/${shopId}`;

    await axios
        .request({
            method: "GET",
            url: imageKitUrl,
            headers: {
                Authorization: basicAuth,
            },
        })
        .then(({ data }) => {
            res.status(200).json({ data });
        })
        .catch(({ err }) => {
            res.status(400).json({ err });
        });
};
```

I have checked that my Vercel deployment has the proper environment variables.

(Look for the `400` response in the network tab at https://accomplice.ink/shops/cs-tattoo)

### Example

https://accomplice.ink/shops/cs-tattoo

**Comment by leerob** [comment, marked as answer]
Hey there! A few notes:

- You don't need to use `axios`, you can use the default Web `fetch` API
- Since this code runs on the server, you wouldn't want `NEXT_PUBLIC_*` since that would expose it to the client-side. Instead, remove that prefix.

```jsx
export default async handler(req, res) => {
    const query = req.query;
    const { shopId } = query;

    try {
        const response = await fetch(`https://api.imagekit.io/v1/files?path=/shops/${shopId}`, {
            method: "GET",
            headers: {
                Authorization: `Basic ${process.env.IMAGEKIT_PRIVATE_KEY}`,
            },
        });

        if (!response.ok) {
            throw new Error("Failed to fetch data from the server");
        }

        const data = await response.json();
        res.status(200).json({ data });
    } catch (err) {
        res.status(400).json({ err });
    }
};
```

What do the Vercel logs say? https://vercel.com/docs/concepts/observability/runtime-logs

**Reply by christina-de-martinez** [reply]
Hey @leerob - thanks for the advice! 

> What do the Vercel logs say? https://vercel.com/docs/concepts/observability/runtime-logs

The build logs look normal. The runtime logs show that 400 error.

<img width="588" alt="Screen Shot 2023-04-10 at 11 30 56 AM" src="https://user-images.githubusercontent.com/31422523/230968488-67e6d483-4313-4c80-8e9b-eb3f2fd28088.png">

**Reply by christina-de-martinez** [reply]
Update: removing that `NEXT_PUBLIC` prefix fixed my issue. Thank you so much @leerob!

**Reply by liMingHui-00** [reply]
Sorry to bother you, what is the purpose of NEXT-PUBLIC and why does deleting NEXT-PUBLIC solve the problem.
